src/renderer/vdom/set-accessor.ts

```diff
@@ -1,49 +1,39 @@
 import * as d from '../../declarations';
 import { Build } from '../../util/build-conditionals';
 import { elementHasProperty } from '../../core/proxy-members';
-import { EMPTY_ARR, EMPTY_OBJ, PROP_TYPE } from '../../util/constants';
+import { PROP_TYPE } from '../../util/constants';
 import { toLowerCase } from '../../util/helpers';
 import { updateAttribute } from './update-attribute';
 
 
-export function setAccessor(plt: d.PlatformApi, elm: any, memberName: string, oldValue: any, newValue: any, isSvg?: boolean, isHostElement?: boolean, i?: any, ilen?: number, cmpMeta?: d.ComponentMeta) {
+export function setAccessor(plt: d.PlatformApi, elm: HTMLElement, memberName: string, oldValue: any, newValue: any, isSvg: boolean, isHostElement: boolean) {
   if (memberName === 'class' && !isSvg) {
     // Class
     if (oldValue !== newValue) {
-      const oldList: string[] = (oldValue == null || oldValue === '') ? EMPTY_ARR : oldValue.trim().split(/\s+/);
-      const newList: string[] = (newValue == null || newValue === '') ? EMPTY_ARR : newValue.trim().split(/\s+/);
+      const oldList = parseClassList(oldValue);
+      const newList = parseClassList(newValue);
+      const toRemove = oldList.filter(item => !newList.includes(item));
 
-      let classList: string[] = (elm.className == null || elm.className === '') ? EMPTY_ARR : elm.className.trim().split(/\s+/);
+      elm.className = [
+        // delete from classList
+        ...parseClassList(elm.className).filter(item => !toRemove.includes(item)),
 
-      for (i = 0, ilen = oldList.length; i < ilen; i++) {
-        if (newList.indexOf(oldList[i]) === -1) {
-          classList = classList.filter((c: string) => c !== oldList[i]);
-        }
-      }
-
-      for (i = 0, ilen = newList.length; i < ilen; i++) {
-        if (oldList.indexOf(newList[i]) === -1) {
-          classList = [...classList, newList[i]];
-        }
-      }
+        // add classes from the new list but not in old list
+        ...newList.filter(item => !oldList.includes(item))
+      ].join(' ');
 
-      elm.className = classList.join(' ');
     }
 
   } else if (memberName === 'style') {
-    // Style
-    oldValue = oldValue || EMPTY_OBJ;
-    newValue = newValue || EMPTY_OBJ;
-
-    for (i in oldValue) {
-      if (!newValue[i]) {
-        (elm as any).style[i] = '';
+    for (const prop in oldValue) {
```

Comment:
Not using i, since the internal type of `i` is a number, not a string. While this is valid JS, it prevents V8 from inlining